src/poetry/puzzle/provider.py

```diff
@@ -546,35 +546,53 @@ def complete_package(
             package = dependency_package.package
             dependency = dependency_package.dependency
             requires = package.all_requires
-        elif package.source_type not in {
-            "directory",
-            "file",
-            "url",
-            "git",
-        }:
-            try:
-                dependency_package = DependencyPackage(
-                    dependency,
-                    self._pool.package(
-                        package.pretty_name,
-                        package.version,
-                        extras=list(dependency.extras),
-                        repository=dependency.source_name,
-                    ),
-                )
-            except PackageNotFound as e:
+        else:
+            if package.source_type == "file":
+                if len(package.files) > 1:
+                    # a locked package might contain further files
+                    # for multiple-constraints dependencies -> prune
+                    path = urllib.parse.urlsplit(package.source_url).path
```

Comment:
I'm guessing `urllib` isn't properly typed?

Reply:
It returns bytes if you feed it with bytes.